pkg/pillar/base/logobjecttypes.go

```diff
@@ -0,0 +1,291 @@
+// Copyright (c) 2020 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package base
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/satori/go.uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+// LogObjectType : Predefined log object types
+type LogObjectType string
+
+// The following is NOT an extensive list.
+// Intent is to create a reference list for engineers to get an idea of its usage.
+const (
+	UnknownLogType         LogObjectType = ""
+	LogType                LogObjectType = "log"
+	ImageLogType           LogObjectType = "image"
+	RelationLogType        LogObjectType = "relation"
+	AppInstanceLogType     LogObjectType = "app_instance"
+	NetworkInstanceLogType LogObjectType = "network_instance"
+)
+
+// LogObject : Holds all key value pairs to be logged later.
+type LogObject struct {
+	Initialized bool
+	Fields      map[string]interface{}
+}
+
+var logObjectMap = make(map[string]*LogObject)
+
+// LoggableObject :
+type LoggableObject interface {
+	Key() string
+	LogCreate()
+	LogModify(old interface{})
+	LogDelete()
+}
+
+// NewLogObject :
+// objType -> log, action, relation, app_instance, image etc
+// objName -> App instance name, name of file being downloaded etc
+// objUUID -> UUID of the object if present (App instance UUID) or UUID of parent object
+// keys -> Keys on which GetValue is called to get values
```

Comment:
Do you expect objName and keys to be different? Note that the argument is "key" not "keys"

Maybe objUUID isn't needed - if objType+objName are unique.

Reply:
Yes objName and key can be different. In the AppInstanceStatus key is UUID and objName can be the name of app instance.
Idea of having objIUUID is for easy filtering of life cycle of events/logs. Can be used to get all the life cycle events with a given objUUID etc.

Reply:
That sounds like objName is a user-friendly name and UUID is what's unique.
But the problem is that we have objects which do not have a UUID.

Thus if we want to I'd call them 
displayName string // Not necessarily unique 
key string // unique - a UUID if that is the unique identifier for the object